cpp/src/parquet/arrow/reader_internal.cc

```diff
@@ -178,16 +685,13 @@ std::shared_ptr<ChunkedArray> CastChunksTo(
 }
 
 Status TransferDictionary(RecordReader* reader,
-                          const std::shared_ptr<DataType>& logical_value_type,
+                          std::shared_ptr<DataType> logical_value_type,
```

Comment:
I found this change from passing by const ref to value threw me a bit.  Could warrant a comment?  Or does it make sense to keep the signature as accepting a const ref and move this:
```
::arrow::dictionary(::arrow::int32(), logical_value_type)
``` 
into `TransferDictionary`?

Reply:
I'm reverting this change, it was a leftover from mid-way through the patch